src/it/java/teammates/it/ui/webapi/GetSessionResultsActionIT.java

```diff
@@ -0,0 +1,353 @@
+
+        ______TS("Accessible for authenticated instructor when published");
+        submissionParams = new String[] {
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, publishedFeedbackSession.getName(),
+                Const.ParamsNames.COURSE_ID, course.getId(),
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_RESULT.name(),
+        };
+        verifyAccessibleForInstructorsOfTheSameCourse(course, submissionParams);
+        verifyInaccessibleForInstructorsOfOtherCourses(course, submissionParams);
+
+        ______TS("Accessible for authenticated student when published");
+        submissionParams = new String[] {
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, publishedFeedbackSession.getName(),
+                Const.ParamsNames.COURSE_ID, course.getId(),
+                Const.ParamsNames.INTENT, Intent.STUDENT_RESULT.name(),
+        };
+        verifyAccessibleForStudentsOfTheSameCourse(course, submissionParams);
+        verifyInaccessibleForStudentsOfOtherCourse(course, submissionParams);
+
+        ______TS("Invalid intent");
+        submissionParams = new String[] {
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, publishedFeedbackSession.getName(),
+                Const.ParamsNames.COURSE_ID, publishedFeedbackSession.getCourse().getId(),
+                Const.ParamsNames.INTENT, Intent.INSTRUCTOR_SUBMISSION.name(),
+        };
+        verifyHttpParameterFailure(submissionParams);
+        submissionParams = new String[] {
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, publishedFeedbackSession.getName(),
+                Const.ParamsNames.COURSE_ID, publishedFeedbackSession.getCourse().getId(),
+                Const.ParamsNames.INTENT, Intent.STUDENT_SUBMISSION.name(),
+        };
+        verifyHttpParameterFailure(submissionParams);
+    }
+
+    private boolean isSessionResultsDataEqual(SessionResultsData self, SessionResultsData other) {
+        List<SessionResultsData.QuestionOutput> thisQuestions = self.getQuestions();
+        List<SessionResultsData.QuestionOutput> otherQuestions = other.getQuestions();
+        if (thisQuestions.size() != otherQuestions.size()) {
+            return false;
+        }
+        for (int i = 0; i < thisQuestions.size(); i++) {
+            SessionResultsData.QuestionOutput thisQuestion = thisQuestions.get(i);
+            SessionResultsData.QuestionOutput otherQuestion = otherQuestions.get(i);
+            if (!isQuestionOutputEqual(thisQuestion, otherQuestion)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean isQuestionOutputEqual(SessionResultsData.QuestionOutput self,
+                                          SessionResultsData.QuestionOutput other) {
+        if (!JsonUtils.toJson(self.getFeedbackQuestion()).equals(JsonUtils.toJson(self.getFeedbackQuestion()))
+                || !self.getQuestionStatistics().equals(other.getQuestionStatistics())) {
```

Comment:
could i check why                 || self.getHasResponseButNotVisibleForPreview() != other.getHasResponseButNotVisibleForPreview()
                || self.getHasCommentNotVisibleForPreview() != other.getHasCommentNotVisibleForPreview()) is not being checked here, like in the old test?

Reply:
Thanks for the catch! They are supposed to be tested. The preview functionality was not in the v9-migration code before the recent merge from master, and I seem to have missed some things when migrating the code added by the merge. Will add these checks in and check the rest of the code again in case I missed anything else.